### ISSUE TITLE
[Bug Report] invalid softplus backward result

### ISSUE THREAD

**Issue by hschoi4448** [tenstorrent-metal/tt-metal#6598]
**Describe the bug**

The `softplus_bw` function returns an invalid gradient value.

**To Reproduce**
1. Copy and past below code to /tests/tt_eager/python_api_testing/unit_testing/backward_ops/test_backward_softplus.py
```Python
# SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.

# SPDX-License-Identifier: Apache-2.0

import torch
import pytest
import tt_lib
from tests.tt_eager.python_api_testing.unit_testing.backward_ops.utility_funcs import data_gen_pt_tt, compare_results

def data_gen_pt_tt(input_shapes, device, required_grad=False, val=1):
    pt_tensor = (torch.ones(input_shapes, requires_grad=required_grad) * val).bfloat16()
    tt_tensor = (
        tt_lib.tensor.Tensor(pt_tensor, tt_lib.tensor.DataType.BFLOAT16).to(tt_lib.tensor.Layout.TILE).to(device)
    )
    return pt_tensor, tt_tensor


@pytest.mark.parametrize(
    "input_shapes",
    (
        (torch.Size([1, 1, 32, 32])),
        # (torch.Size([1, 1, 320, 384])),
        # (torch.Size([1, 3, 320, 384])),
    ),
)
@pytest.mark.parametrize(
    "beta",
    # [0.5, -3, 1, 4],
    [1]
)
@pytest.mark.parametrize(
    "threshold",
    # [-20, -10, 10, 20, 5],
    [0]
)
def test_bw_softplus(input_shapes, beta, threshold, device):
    in_data, input_tensor = data_gen_pt_tt(input_shapes, device, True, val=1)
    grad_data, grad_tensor = data_gen_pt_tt(input_shapes, device, False, val=10)

    print("input_tensor", input_tensor)
    print("grad_tensor", grad_tensor)
    
    in_data.retain_grad()

    pyt_y = torch.nn.functional.softplus(in_data, beta=beta, threshold=threshold)

    tt_output_tensor_on_device = tt_lib.tensor.softplus_bw(grad_tensor, input_tensor, beta=beta, threshold=threshold)

    pyt_y.backward(gradient=grad_data)

    golden_tensor = [in_data.grad]

    comp_pass = compare_results(tt_output_tensor_on_device, golden_tensor)
    print("tt_output_tensor_on_device", tt_output_tensor_on_device)
    print("golden_tensor", golden_tensor)
    assert comp_pass
```
2. Run cmd `pytest ./tests/tt_eager/python_api_testing/unit_testing/backward_ops/test_backward_softplus.py`
```Python
input_tensor ttnn.Tensor([[[[ 1.00000,  1.00000,  ...,  1.00000,  1.00000],
               [ 1.00000,  1.00000,  ...,  1.00000,  1.00000],
               ...,
               [ 1.00000,  1.00000,  ...,  1.00000,  1.00000],
               [ 1.00000,  1.00000,  ...,  1.00000,  1.00000]]]], shape=Shape([1, 1, 32, 32]), dtype=DataType::BFLOAT16, layout=Layout::TILE)
grad_tensor ttnn.Tensor([[[[10.00000, 10.00000,  ..., 10.00000, 10.00000],
               [10.00000, 10.00000,  ..., 10.00000, 10.00000],
               ...,
               [10.00000, 10.00000,  ..., 10.00000, 10.00000],
               [10.00000, 10.00000,  ..., 10.00000, 10.00000]]]], shape=Shape([1, 1, 32, 32]), dtype=DataType::BFLOAT16, layout=Layout::TILE)
tt_output_tensor_on_device [ttnn.Tensor([[[[ 7.21875,  7.21875,  ...,  7.21875,  7.21875],
               [ 7.21875,  7.21875,  ...,  7.21875,  7.21875],
               ...,
               [ 7.21875,  7.21875,  ...,  7.21875,  7.21875],
               [ 7.21875,  7.21875,  ...,  7.21875,  7.21875]]]], shape=Shape([1, 1, 32, 32]), dtype=DataType::BFLOAT16, layout=Layout::TILE)]
golden_tensor [tensor([[[[10., 10., 10.,  ..., 10., 10., 10.],
          [10., 10., 10.,  ..., 10., 10., 10.],
          [10., 10., 10.,  ..., 10., 10., 10.],
          ...,
          [10., 10., 10.,  ..., 10., 10., 10.],
          [10., 10., 10.,  ..., 10., 10., 10.],
          [10., 10., 10.,  ..., 10., 10., 10.]]]], dtype=torch.bfloat16)]
```

**Expected behavior**

I want `softplus_bw` to return the correct gradient

**Screenshots**

**Please complete the following environment information:**
- OS: [e.g. Ubuntu 20.04]
- Version of software (eg. commit fcd006b981fa1b0d8a0f7395060513d4e402a7a9)
- Device: wormhole_b0

**Additional context**
```C++
        [beta, threshold](scalar_t a, scalar_t b) -> scalar_t {
          float z = std::exp(float(b) * beta);
          return (float(b) * beta) > threshold ? a : static_cast<scalar_t>(float(a) * z / (z + float(1.)));
        },
```
https://github.com/pytorch/pytorch/blob/main/aten/src/ATen/native/cpu/Activation.cpp#L1095

In my opinion, it seems like this code is the correct implementation.

**Comment by VirdhatchaniKN** [comment]
Hi @eyonland , @hschoi4448 
Provided fix for backward softplus : #6675 
Kindly review, Thanks .

**Comment by VirdhatchaniKN** [comment]
Merged changes #6675
